event/handler.go

```diff
@@ -84,6 +87,15 @@ func (handler *ExportHandler) Handle(event *FilterSubmitted) error {
 
 	fileURL, err := handler.fileStore.PutFile(reader, filter)
 	if err != nil {
+		if strings.Contains(err.Error(), observation.ErrNoResultsFound.Error()) {
```

Comment:
Should we be doing this based on the Reader results? If this only happens if uploading the file fails, it seems that we won't set the 'empty' state quite a lot of the time. 

Should we be uploading empty files?

Do we need to do something else to prevent the xls event being queued?

Reply:
I added a comment explaining that the neo4j stream is only read when the S3 library is reading the data. This will result in no file being uploaded and no event is sent to the xlsx as this function returns before reaching that point.